src/video/windows/SDL_windowsshape.c

```diff
@@ -37,10 +37,12 @@ Win32_CreateShaper(SDL_Window * window) {
     result->driverdata = (SDL_ShapeData*)SDL_malloc(sizeof(SDL_ShapeData));
     ((SDL_ShapeData*)result->driverdata)->mask_tree = NULL;
     /* Put some driver-data here. */
-    window->shaper = result;
```

Comment:
Did you mean to move this before `return result;` ?

Reply:
I think the following is needed, on top of existing patch: `Win32_ResizeWindowShape`
assumes a legitimate shaper pointer.
```diff
diff --git a/src/video/windows/SDL_windowsshape.c b/src/video/windows/SDL_windowsshape.c
index 24f125b..7d15f4d 100644
--- a/src/video/windows/SDL_windowsshape.c
+++ b/src/video/windows/SDL_windowsshape.c
@@ -39,2 +39,3 @@ Win32_CreateShaper(SDL_Window * window) {
     /* Put some driver-data here. */
+    window->shaper = result;
     resized_properly = Win32_ResizeWindowShape(window);
@@ -43,2 +44,3 @@ Win32_CreateShaper(SDL_Window * window) {
         SDL_free(result);
+        window->shaper = NULL;
         return NULL;
```

Reply:
The caller of CreateShaper assigns the return value to window->shaper.

Reply:
Oh wait, I see, you're right.

Reply:
ok, forced-pushed this correction, thanks!